src/github/task-workflow.ts

```diff
@@ -0,0 +1,220 @@
+import { Task } from '../tasks';
+import { GitHub } from './github';
+import { GithubWorkflow } from './workflows';
+import { ContainerOptions, Job, JobPermissions, JobStep, Triggers } from './workflows-model';
+
+const DEFAULT_JOB_ID = 'build';
+const UBUNTU_LATEST = 'ubuntu-latest';
+
+export interface TaskGithubWorkflowOptions {
+  /**
+   * The workflow name.
+   */
+  readonly name: string;
+
+  /**
+   * The primary job id.
+   * @default "build"
+   */
+  readonly jobId?: string;
+
+  /**
+   * @default - default image
+   */
+  readonly container?: ContainerOptions;
+
+  /**
+   * Adds an 'if' condition to the workflow.
+   */
+  readonly condition?: string;
+
+  /**
+   * A directory name which contains artifacts to be uploaded (e.g. `dist`).
+   *
+   * @default - not set
+   */
+  readonly artifactsDirectory?: string;
+
+  /**
+   * The triggers for the workflow.
+   *
+   * @default - by default workflows can only be triggered by manually.
+   */
+  readonly trigger?: Triggers;
+
+  /**
+   * Initial steps to run before the source code checkout.
+   *
+   * @default - not set
+   */
+  readonly preCheckoutSteps?: JobStep[];
+
+  /**
+   * Override for the `with` property of the source code checkout step.
+   *
+   * @default - not set
+   */
+  readonly checkoutWith?: Record<string, any>;
+
+  /**
+   * Steps to run before the main build step.
+   *
+   * @default - not set
+   */
+  readonly preBuildSteps?: JobStep[];
+
+  /**
+   * The main task to be executed.
+   */
+  readonly task: Task;
+
+  /**
+   * Main build step used in the workflow.
+   *
+   * @default - by default we will run `projen ${task.name}`
+   */
+  readonly buildStep?: JobStep;
+
+  /**
+   * Actions to run after the main build step.
+   *
+   * @default - not set
+   */
+  readonly postBuildSteps?: JobStep[];
+
+  /**
+   * Enables anti-tamper checks in the workflow.
+   *
+   * @default true
+   */
+  readonly antitamper?: boolean;
+
+  /**
+   * Actions to run as the last step in the job.
+   *
+   * @default - not set
+   */
+  readonly finalSteps?: JobStep[];
+
+  /**
+   * Workflow environment variables.
+   * @default {}
+   */
+  readonly env?: Record<string, string>;
+
+  /**
+   * Permissions for the build job.
+   */
+  readonly permissions: JobPermissions;
+}
+
+/**
+ * A GitHub workflow for common build tasks within a project.
+ */
+export class TaskGithubWorkflow extends GithubWorkflow {
```

Comment:
We should rename `GithubWorkflow` to `Workflow` at some point (because now it's under the `github` submodule), but this one can already be named appropriately:

```suggestion
export class TaskWorkflow extends GithubWorkflow {
```